client/src/app/admin/panel/settings/admin-panel-advanced-settings.js

```diff
@@ -156,24 +162,47 @@ class AdminPanelAdvancedSettings extends React.Component {
             <Form className="admin-panel-advanced-settings__api-keys-modal" onSubmit={this.addAPIKey.bind(this)}>
                 <Header title={i18n('ADD_API_KEY')} description={i18n('ADD_API_KEY_DESCRIPTION')}/>
                 <FormField name="name" label={i18n('NAME_OF_KEY')} validation="DEFAULT" required fieldProps={{size: 'large'}}/>
-                <SubmitButton type="secondary">{i18n('SUBMIT')}</SubmitButton>
+                <div className="admin-panel-advanced-settings__api-keys__permissions">
+                    <FormField className = "admin-panel-advanced-settings__api-keys__permissions__item" name="createTicketPermission" label={i18n('TICKET_CREATION_PERMISSION')}   field='checkbox'/>
+                    <FormField name="ticketNumberPermission" label={i18n('TICKET_NUMBER_RETURN_PERMISSION')}  field='checkbox'/>
+                </div>
+                <div className="admin-panel-advanced-settings__api-keys__permissions" >
+                    <FormField className = "admin-panel-advanced-settings__api-keys__permissions__item" name="commentTicketPermission" label={i18n('TICKET_COMMENT_PERMISSION')}  field='checkbox'/>
+                    <FormField name="userPermission" label={i18n('USER_CREATION_PERMISSION')}   field='checkbox'/>
+                </div>
+                <SubmitButton className="admin-panel-advanced-settings__api-keys-modal__submit-button" type="secondary">{i18n('SUBMIT')}</SubmitButton>
             </Form>
         );
     }
 
-    addAPIKey({name}) {
+    addAPIKey({name,userPermission,createTicketPermission,commentTicketPermission,ticketNumberPermission}) {
         ModalContainer.closeModal();
+        this.setState({
+            error: ''
+        })
```

Comment:
Remember that calls to `setState` are asynchronous, so don't expect the next `API.call` to run after the state is set. In this situation, the `API.call` is obviously also asynchronous, so I don't think the risk is considerable, but it'd be better to wrap the API.call as the callback of this `setState`. @ivandiazwm what do you think about this?

Reply:
@mredigonda I think it's ok. If the state doesn't update before `API.call` finished, it will be overwritten by the next setState if it fails. We use the same pattern for loading.